pkg/apis/tensorflow/v1/types.go

```diff
@@ -69,6 +69,9 @@ type TFJobSpec struct {
 	//     "Worker": ReplicaSpec,
 	//   }
 	TFReplicaSpecs map[TFReplicaType]*common.ReplicaSpec `json:"tfReplicaSpecs"`
+
+	// A switch to enable dynamic worker
+	EnableDynamicWorker bool `json:"enableDynamicWorker,omitempty"`
```

Comment:
Since it is a optional field, can we use *bool?

```
	EnableDynamicWorker *bool `json:"enableDynamicWorker,omitempty"`
```

Reply:
The zero value of a boolean variable is `false`, which, as you can see in the following PRs, works perfectly for the default case and is back-compatible. IMHO, I'll try to use the pointer type as less as possible because we don't have to worry about `nil` anywhere. Only when I need to differentiate the zero value vs not-set value, would I use the pointer type.

Reply:
I prefer not to rely on the zero value. The value will be set to default (false here) by the mutating webhook in the future. If we do not use a pointer, it is hard to determine if the value is set.